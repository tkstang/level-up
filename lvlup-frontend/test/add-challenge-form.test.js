import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import AddChallengeForm from '../src/components/Admin/challenges/add-challenge/add-challenge-form';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('AddChallengeForm should render', () => {
  const store = mockStore({ lvlupApp });
  const addchallenge = shallow(
    <AddChallengeForm campuses={{ length: 5 }} store={store} />,
  );
  expect(shallowToJson(addchallenge)).toMatchSnapshot();
});
