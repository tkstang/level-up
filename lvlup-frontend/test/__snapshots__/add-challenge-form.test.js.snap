exports[`test AddChallengeForm should render 1`] = `
<ReduxForm
  addChallenge={[Function]}
  allCampuses={[Function]}
  setCampuses={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  } />
`;
